typegraph/core/src/utils/mod.rs

```diff
@@ -38,6 +45,49 @@ fn find_missing_props(
     Ok(missing_props)
 }
 
+struct Oauth2Params {
+    name: String,
+    authorize_url: String,
+    access_url: String,
+    scopes: String,
+    profile_url: Option<String>,
+    profiler: Option<TypeId>,
+}
+
+fn gen_oauth2(params: Oauth2Params) -> Auth {
+    let mut auth_data = IndexMap::new();
+
+    auth_data.insert(
+        "authorize_url".to_string(),
+        serde_json::to_value(params.authorize_url).unwrap(),
+    );
+    auth_data.insert(
+        "access_url".to_string(),
+        serde_json::to_value(params.access_url).unwrap(),
+    );
+    auth_data.insert(
+        "scopes".to_string(),
+        serde_json::to_value(params.scopes).unwrap(),
+    );
+    auth_data.insert(
+        "profile_url".to_string(),
+        serde_json::to_value(params.profile_url).unwrap(),
+    );
+    auth_data.insert(
+        "profiler".to_string(),
+        params
+            .profiler
+            .map(|p| p.into())
+            .unwrap_or(serde_json::Value::Null),
+    );
```

Comment:
Could be simpler and easier to read with the `json!` macro, `serde_json::from_value::<IndexMap<String, serde_json::Value>>(...)`